enterprise_access/apps/api/v1/views.py

```diff
@@ -746,7 +746,128 @@ def partial_update(self, request, *args, **kwargs):
+        """
+        Args:
+            policy_uuid: a list of one or more valid uuid strings
+        Raises:
+            SubsidyAccessPolicyError: if policy uuid is invalid or there is an attempt to change policy uuid.
+        Returns:
+            policy instance
+        """
+        if policy_uuid:
+            try:
+                instance = SubsidyAccessPolicy.objects.get(uuid=policy_uuid)
+            except ObjectDoesNotExist as exc:
+                error_msg = f'Policy UUID provided ({policy_uuid}) is not a valid UUID'
+                logger.exception(error_msg)
+                raise SubsidyAccessPolicyError(error_msg, status.HTTP_400_BAD_REQUEST) from exc
+            # for updates, ensure that uuid is not editable
+            if instance and policy_data.get('uuid') != str(instance.uuid):
+                error_msg = f'Policy UUID cannot be changed'
+                logger.exception(error_msg)
+                raise SubsidyAccessPolicyError(error_msg, status.HTTP_400_BAD_REQUEST)
+            return instance
+
+    def partial_update(self, request, *args, **kwargs):
+        """
+        Used for http patch method. Updates the policy data passed in request after validation.
+        """
+        policy_data = request.data
+        policy_uuid = self.kwargs.get('uuid')
+        try:
+            instance = self._validate_policy(policy_uuid, policy_data)
+        except SubsidyAccessPolicyError as exc:
+            logger.exception(exc)
+            return Response(exc.message, exc.http_status_code)
+        serializer = self.get_serializer(instance, data=policy_data, partial=True)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_202_ACCEPTED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def list(self, request, *args, **kwargs):
+        """
+        List action for SubsidyAccessPolicyCRUDViewset. Show all policies linked with the group uuid passed in request.
+        """
+        group_uuid = self.requested_group_uuid
+        if not group_uuid:
+            msg = '"group_uuid" query param is required'
+            return Response(msg, status=status.HTTP_400_BAD_REQUEST)
+        return super().list(request, *args, **kwargs)
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Retrieve action for SubsidyAccessPolicyCRUDViewset
+        """
+        return super().retrieve(request, *args, **kwargs)
+
+    def get_queryset(self):
+        """
+        Queryset for the SubsidyAccessPolicyCRUDViewset. The admins should only be able to perform CRUD operations
+        on the enterprise's policies they belong to.
+        """
+        queryset = SubsidyAccessPolicy.objects.order_by('-created')
+        group_uuid = self.requested_group_uuid
+        if group_uuid:
+            queryset = queryset.filter(group_uuid=group_uuid)
+        return queryset
```

Comment:
`requests.has_admin_access` permission will be checked before the view code is executed? If yes, then I think we will always get `group_uuid` because `group_uuid` is required for permission code + others. If yes, then I think we can remove `if group_uuid:` above.